protocol/xmrtaker/swap_state.go

```diff
@@ -452,29 +451,21 @@ func (s *swapState) lockETH(amount common.EtherAmount) (ethcommon.Hash, error) {
 	if s.ethAsset != types.EthAssetETH {
 		// TODO: check logs
 		// TODO: check units
-
-		// check that the approval is required
 		// TODO: separate to its own function and create unit tests
 		token, err := contracts.NewIERC20(s.ethAsset.Address(), s.EthClient())
 		if err != nil {
-			log.Errorf("failed to instantiate IERC20: %s", s.ethAsset)
-			return ethcommon.Hash{}, err
+			return ethcommon.Hash{}, fmt.Errorf("failed to instantiate IERC20: %s", err)
 		}
-		allowance, err := token.Allowance(s.CallOpts(), s.ethAsset.Address(), s.ContractAddr())
+
+		balance, err := token.BalanceOf(s.CallOpts(), s.EthAddress())
 		if err != nil {
-			log.Errorf("failed to get allowance for token: %s", s.ethAsset)
-			return ethcommon.Hash{}, err
+			return ethcommon.Hash{}, fmt.Errorf("failed to get balance for token: %s", err)
 		}
 
-		if allowance.Cmp(amount.BigInt()) == -1 {
-			log.Info("approving token for use by the swap contract...")
-			_, _, err = s.sender.Approve(s.ContractAddr(), amount.BigInt())
-			if err != nil {
-				log.Errorf("failed to approve token: %s", s.ethAsset)
-				return ethcommon.Hash{}, err
-			}
-		} else {
-			log.Info("the token has already been approved, continuing...")
+		log.Info("approving token for use by the swap contract...")
```

Comment:
If we are using an ellipsis to indicate that the action may take a long time, do we need a log saying that it was approved afterwards?

Reply:
yeah, I added a log afterwards